# A/B Testing Analysis — Website Background Experiment


## Objective
Evaluate whether changing website background color black to white improves conversion rate without harming engagement.

Group A = White color (Control)

Group B = Black color (treatment)

data_source: https://www.kaggle.com/datasets/adarsh0806/ab-testing-practice/data

In [3]:

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from statsmodels.stats.proportion import proportions_ztest, confint_proportions_2indep


## Load Data

In [4]:

df = pd.read_csv('ab_testing.csv')
df.head()


,User ID,Group,Page Views,Time Spent,Conversion,Device,Location
0,14292,B,3,424,No,Mobile,Northern Ireland
1,11682,A,9,342,No,Mobile,Scotland
2,19825,A,2,396,No,Desktop,Northern Ireland
3,16080,B,4,318,No,Desktop,Wales
4,18851,A,1,338,Yes,Desktop,Scotland


 Data Preparation

In [5]:

df['converted'] = (df['Conversion'] == 'Yes').astype(int)
df.head()


,User ID,Group,Page Views,Time Spent,Conversion,Device,Location,converted
0,14292,B,3,424,No,Mobile,Northern Ireland,0
1,11682,A,9,342,No,Mobile,Scotland,0
2,19825,A,2,396,No,Desktop,Northern Ireland,0
3,16080,B,4,318,No,Desktop,Wales,0
4,18851,A,1,338,Yes,Desktop,Scotland,1


##  Summary Metrics

In [6]:

summary = df.groupby('Group').agg(
    users=('User ID', 'count'),
    conversions=('converted', 'sum'),
    conversion_rate=('converted', 'mean'),
    avg_time_spent=('Time Spent', 'mean'),
    avg_page_views=('Page Views', 'mean')
)

summary


,users,conversions,conversion_rate,avg_time_spent,avg_page_views
Group,,,,,
A,2519,136,0.053990,241.733227,7.581580
B,2481,349,0.140669,243.303910,7.492946


Statistical Test (Z-test)

In [8]:

conv = summary['conversions'].values
nobs = summary['users'].values

z_stat, p_value = proportions_ztest(conv, nobs)

print("Z-stat:", z_stat)
print("P-value:", p_value)


Z-stat: -10.354488520266216
P-value: 3.9931511894201938e-25


## 📏 Confidence Interval

In [9]:

low, high = confint_proportions_2indep(
    count1=conv[1], nobs1=nobs[1],
    count2=conv[0], nobs2=nobs[0],
    method='wald'
)

print("CI:", (low, high))


CI: (np.float64(0.07039888441690295), np.float64(0.1029599287881478))



##  Sumary

The treatment variant significantly improves conversion rate by +8.67% (95% CI: +7.0% to +10.3%, p < 0.001), with no negative impact on engagement metrics, supporting a full rollout.
